### PR TITLE
Add Http3Transport

### PR DESCRIPTION
**Pull request by pthatcherg**
Replace PooledHttpTransport and reference https://tools.ietf.org/html/draft-vvv-webtransport-http3-00

### CODE REVIEW

File: explainer.md [hunk 1]
```diff
@@ -253,10 +268,11 @@ A QuicTransport is a WebTransport that maps directly to QUIC streams and
...
 
-A PooledHttpTransport is a WebTransport that provides different subset of these
-capabilities depending on the underlying HTTP protocol (HTTP/3 providing the
-widest support).  It has the advantages that HTTP and non-HTTP traffic can share
-the same network port and congestion control context.
+An Http3Transport is a WebTransport that provides QUIC streams and datagrams
+with slightly more overhead vs. a QuicTransport.  It has the advantage that HTTP
+and non-HTTP traffic can share the same network port and congestion control
+context, and it may be pooled with other transports such that the transport may
+be connected more quickly (by resuing an existing HTTP/3 connection).
```
Comment (shampson) [inline explainer.md]: resuing --> reusing

File: index.html [hunk 2]
```diff
@@ -1811,6 +1811,125 @@ <h3>Interface <dfn>ReceiveStream</dfn></h3>
...
+              internal slot representing a sequence of <code><a>OutgoingStream</a></code>
+              objects, initialized to empty.
+            </li>
+            <li>Let <var>transport</var> have a <dfn>[[\IncomingStreams]]</dfn>
+              internal slot representing a sequence of <code><a>IncomingStream</a></code>
+              objects, initialized to empty.
+            </li>
+            <li>
+                Let <var>transport</var> have a <dfn>[[\WebTransportState]]</dfn>
+                internal slot, initialized to <code>"connecting"</code>.
```
Comment (shampson) [inline index.html]: Is the "connecting" redundant? You mentioned it above that the transport object has a "connecting" state.

File: index.html [hunk 3]
```diff
@@ -1811,6 +1811,125 @@ <h3>Interface <dfn>ReceiveStream</dfn></h3>
...
+            </li>
+            <li>Let <var>transport</var> have a <dfn>[[\ReceivedDatagrams]]</dfn>
+                internal slot representing a queue of <code>Uint8Array</code>, initialized to empty.
+            </li>
+            <li>Let <var>transport</var> have a <dfn>[[\ReceiveDatagramsPromise]]</dfn>
```
Comment (shampson) [inline index.html]: What about mentioning the ReadyToSendDatagramsPromise? This isn't included in the QuicTransport either. We also don't mention these internal slots in the methods which should probably be done.

File: index.html [hunk 4]
```diff
@@ -1811,6 +1811,125 @@ <h3>Interface <dfn>ReceiveStream</dfn></h3>
...
+                internal slot representing a <code>Promise&lt;sequence&lt;Uint8Array&gt;&gt;?</code>,
+                initialized to null.
+            <li>Run these steps in parallel:
+              <ol>
+                <li>
+                  Either establish an HTTP/3 connection or reuse an existing HTTP/3
```
Comment (shampson) [inline index.html]: Do we need to specify how we establish an HTTP/3 connection or reuse one here?

File: explainer.md (hunk 1 above)
Comment (pthatcherg) [reply explainer.md]: Fixed

File: index.html (hunk 4 above)
Comment (pthatcherg) [reply index.html]: I'll punt to the IETF doc :)

File: index.html (hunk 2 above)
Comment (pthatcherg) [reply index.html]: Good catch; removed

File: index.html (hunk 3 above)
Comment (pthatcherg) [reply index.html]: Let's make it an issue to follow up on

**Pull request merged**